core/templates/services/user-backend-api.service.ts

```diff
@@ -136,19 +132,54 @@ export class UserBackendApiService {
     }).toPromise();
   }
 
-  async getPreferencesAsync(): Promise<PreferencesBackendDict> {
-    return this.http.get<PreferencesBackendDict>(this.PREFERENCES_DATA_URL)
-      .toPromise();
+  async getPreferencesAsync():
+    Promise<NonEmailPreferencesBackendDict> {
+    return this.http.get<NonEmailPreferencesBackendDict>(
+      this.PREFERENCES_DATA_URL).toPromise();
+  }
+
+  async getPreferencesProfilePictureDataUrlAsync():
+    Promise<ProfilePictureDataBackendDict> {
+    return this.http.get<ProfilePictureDataBackendDict>(
+      this.PREFERENCES_PROFILE_PICTURE_DATA_URL).toPromise();
+  }
+
+  async getEmailPreferencesAsync():
+    Promise<EmailPreferencesBackendDict> {
+    return this.http.get<EmailPreferencesBackendDict>(
+      this.EMAIL_PREFFERENCES).toPromise();
   }
 
   async updatePreferencesDataAsync(
       updateType: string,
-      data: boolean | string | string[] | EmailPreferencesBackendDict
+      data: boolean | string | string[]
+  ): Promise<NonEmailPreferencesBackendDict> {
+    return this.http.put<NonEmailPreferencesBackendDict>(
+      this.PREFERENCES_DATA_URL, {
+        update_type: updateType,
```

Comment:
Why are we sending `update_type`?

Reply:
We currently use the `update_type` argument to know which value will be updated. But I am thinking about using the `data` argument as a dictionary to update values. For that `update_type` will not be needed anymore.

Reply:
Removed the use of `update_type`. Now we are sending all the data at once in a dictionary.